.github/workflows/tests.yml

```diff
@@ -21,6 +21,10 @@ on:
       - "requirements_dev.txt"
       - "requirements.txt"
 
+concurrency: 
+  group: ${{ github.head_ref }}
```

Comment:
Q: Why we group by `github.head_ref` , does it affect , when two PR's are executing the same workflow, does the recent PR cancel the workflow of previous , or they could run parallely.

Reply:
The cancelling of jobs happens only at PR Level right not at workflow file level right?

Reply:
I think ideally we should have it cancel jobs at PR Level

Reply:
https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions
Here it mentions source branch of the pull request.

Reply:
Could you  test it in your forked branch, to see if CI runs are not cancelled for independent PR's.?